gitlab/.gitlab-ci.yml

```diff
@@ -0,0 +1,417 @@
+  DOCKER_TLS_CERTDIR: ""
+
+  # Build Variables
+  BORDEL_REPO: "github.com/apertussolutions/bordel.git"
+  BORDEL_BRANCH: "master"
+  MANIFEST_REPO: "github.com/apertussolutions/openxt-manifest.git"
+  MANIFEST_BRANCH: "master"
+  MANIFEST_FILE: "stable-9.xml"
+  DEBUG: "true"
+  BUILD_TYPE: "all"
+  # Final packaging type: iso, pxe or all(not recommended)
+  PACKAGE_TYPE: "iso"
+  GITLAB_HOST: "<gitlab host>"
+  AWS_REGION: "<region>"
+  S3_BUCKET: "openxt"
+  # If you want the build to fail, switch to building individual jobs. The cache
+  # doesn't work very well in this instance, need to figure something else out
+  # for this.
+  #BUILD_TYPE: "piecemeal"
+
+.setup_cm: &setup_cm
+  - aws configure set default.s3.signature_version s3v4
+  - git config --global user.email "you@example.com"
+  - git config --global user.name "Your Name"
+  - git config --global url.https://${GITLAB_HOST}/.insteadOf git://${GITLAB_HOST}/
+  # This is necessary to be able to pull the source from other gitlab projects.
+  # Instead of remapping the URL in the .gitconfig, do it this way so that the old
+  # access tokens don't get permanently attached to the repository's .git/config
+  - echo "machine ${GITLAB_HOST}" > ~/.netrc
+  - echo "login gitlab-ci-token" >> ~/.netrc
+  - echo "password $CI_JOB_TOKEN" >> ~/.netrc
+
+.default-cache: &default-cache
+  key: $CI_COMMIT_SHA
+  paths:
+    - deploy
+
+.pull-push-cache: &pull-push-cache
+  cache:
+    <<: *default-cache
+    policy: pull-push
+
+.pull-cache: &pull-cache
+  cache:
+    <<: *default-cache
+    policy: pull
+
+setup:docker_build_image:
+  image: docker:18-git
+  services:
+    - docker:18-dind
+  stage: prepare
+  variables:
+    DEBUG: ""
+  before_script: *setup_cm
+  script:
+    # Install the awscli python package and have it execute a docker login
+    - apk -q update && apk -qUu add python py-pip && pip install --quiet awscli
+    - eval $(aws ecr get-login --no-include-email --region ${AWS_REGION} --registry-ids ${ECR_ID})
+    # Clone bordel so it can make the docker image
+    - git clone -b ${BORDEL_BRANCH} https://${BORDEL_REPO}
+    - cd bordel
+    # Remove the ENTRYPOINT, this may not be necessary anymore
+    - sed -i -e 's|ENTRYPOINT.*||' Dockerfiles/openxt-oe64
+    - if [[ "x$DEBUG" != "x" ]]; then docker info; fi
+    # Build the openxt-oe64 Dockerfile, push to ECR
+    - docker build -t ${EC2_DOCKER_REGISTRY}/build_image:debian_jessie -f Dockerfiles/jessie/openxt-oe64 .
+    - docker push ${EC2_DOCKER_REGISTRY}/build_image:debian_jessie
+  when: manual
+  #only:
+  #  - master
+  #  - /^stable-.*/
+
+setup:workspace:
+  image: ${EC2_DOCKER_REGISTRY}/build_image:debian_jessie
+  stage: prepare
+  before_script: *setup_cm
+  script:
+    - if [[ "x$DEBUG" != "x" ]]; then env | sort; df -h; fi
+    # Remove the old cache if present, create the new directory structure
+    - rm -rf {deploy,logs,staging}/* || true
+    - rm -rf openxt-workspace || true
```

Comment:
When working on this, there were instance where the directories didn't exist usually because a previous build didn't complete and some of these directories were missing. Without the `|| true` the `rm` would fail deleting the non-existent directory the build would abort.  I haven't had a chance to revisit this to see if this issue still persists if I remove them.

Reply:
`rm`, when given `-f`, does not fail if asked to remove something that does not exist. Or at least, no `rm` binary that I have seen so far. It would fail with `rm -f <directory>`, but `-r` is present.